### DISCUSSION TITLE
How to change xmake global cache/packages/repositories directory?

### DISCUSSION THREAD

**Discussion by cursey** [xmake-io/xmake#1147] [Q&A] [answered]
As the title says, it's unclear to me how I would go about changing the directory xmake uses to store downloaded and compiled packages. I have a separate drive for development and would like to store all this data on that drive so it isn't taking up space on my C: drive. Is it possible to tweak the directory? If not, could such an option be added?

**Comment by waruqi** [comment, marked as answer]
set XMAKE_GLOBALDIR=D:\yourdir

**Reply by cursey** [reply]
Thanks for the answer. Is this documented somewhere and I just missed it?

**Reply by waruqi** [reply]
You can run `xmake show -l envs` to see a list of all environment variables, but there is no documentation of this yet

**Reply by waruqi** [reply]
You can also use `xmake g --pkg_installdir=xxx` to set the installation package storage path separately, and I just added a new `xmake g --cachedir=xxx` parameter to set the global cache directory separately on dev branch.
